From c53bbe865f1ec6ee5ecad60514a055969e4b35ca Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Tue, 11 Jun 2019 14:54:50 +0800
Subject: [PATCH 2/2] Enable add user button if the dialog canceled before

If the add-user dialog was canceled by clicking outside of dialog,
Then user will not able to create new user anymore since the button
was disabled. Add "onCancel" method to enable the button again.

Change-Id: Ie3200e9d047c4f9542fbca9484d663b47bc7cdc2
Tracked-On: https://jira.devtools.intel.com/browse/OAM-80786
Signed-off-by: Lei,RayX <rayx.lei@intel.com>
---
 .../car/settings/common/ConfirmationDialogFragment.java  | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/com/android/car/settings/common/ConfirmationDialogFragment.java b/src/com/android/car/settings/common/ConfirmationDialogFragment.java
index a9e9e683..92496568 100644
--- a/src/com/android/car/settings/common/ConfirmationDialogFragment.java
+++ b/src/com/android/car/settings/common/ConfirmationDialogFragment.java
@@ -34,7 +34,7 @@ import androidx.fragment.app.DialogFragment;
  * action.
  */
 public class ConfirmationDialogFragment extends DialogFragment implements
-        DialogInterface.OnClickListener {
+        DialogInterface.OnClickListener, DialogInterface.OnCancelListener {

     /** Builder to help construct {@link ConfirmationDialogFragment}. */
     public static class Builder {
@@ -260,6 +260,13 @@ public class ConfirmationDialogFragment extends DialogFragment implements
         }
     }

+    @Override
+    public void onCancel(DialogInterface dialog) {
+        if (mRejectListener != null) {
+            mRejectListener.onReject(getArguments().getBundle(ARGUMENTS_KEY));
+        }
+    }
+
     /** Listens to the confirmation action. */
     public interface ConfirmListener {
         /**
--
2.21.0
